tests/shap/test_log.py

```diff
@@ -0,0 +1,152 @@
+import mlflow
+import shap
+import numpy as np
+import sklearn
+from mlflow.utils import PYTHON_VERSION
+
+
+def test_sklearn_log_explainer():
+    """
+    Tests mlflow.shap log_explainer with mlflow serialization of the underlying model
+    """
+
+    with mlflow.start_run() as run:
+
+        run_id = run.info.run_id
+
+        X, y = shap.datasets.boston()
+        model = sklearn.ensemble.RandomForestRegressor(n_estimators=100)
+        model.fit(X, y)
+
+        explainer_original = shap.Explainer(model.predict, X, algorithm="permutation")
+        shap_values_original = explainer_original(X[:5])
+
+        mlflow.shap.log_explainer(explainer_original, "test_explainer")
+
+        explainer_loaded = mlflow.shap.load_explainer("runs:/" + run_id + "/test_explainer")
+        shap_values_new = explainer_loaded(X[:5])
+
+        np.testing.assert_array_equal(shap_values_original.base_values, shap_values_new.base_values)
+        np.testing.assert_allclose(
+            shap_values_original.values, shap_values_new.values, rtol=100, atol=100
+        )
+
+
+def test_sklearn_log_explainer_self_serialization():
+    """
+    Tests mlflow.shap log_explainer with SHAP internal serialization of the underlying model
+    """
+
+    with mlflow.start_run() as run:
+
+        run_id = run.info.run_id
+
+        X, y = shap.datasets.boston()
+        model = sklearn.ensemble.RandomForestRegressor(n_estimators=100)
+        model.fit(X, y)
+
+        explainer_original = shap.Explainer(model.predict, X, algorithm="permutation")
+        shap_values_original = explainer_original(X[:5])
+
+        mlflow.shap.log_explainer(explainer_original, "test_explainer", False)
```

Comment:
```suggestion
        mlflow.shap.log_explainer(explainer_original, "test_explainer", serialize_model_using_mlflow=False)
```

Can we use a keyword argument here to clarify what `False` represents?